Asp.Versioning.OpenApi@10.0.0 have issues with Microsoft.Extensions.ApiDescription.Server@10.0.10

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

Upgrading from `Asp.Versioning.OpenApi@10.0.0-rc.1` to `Asp.Versioning.OpenApi@10.0.0` causes build errors when using `Microsoft.Extensions.ApiDescription.Server` (only tested with `10.0.10`)

I'm not sure with what package the issue lies, but upgrading Asp.Versioning.OpenApi caused the errors so I'm starting here :)

### Expected Behavior

No build errors

### Steps To Reproduce

add these package to csproj
```xml
<ItemGroup>
  <PackageReference Include="Asp.Versioning.OpenApi" Version="10.0.0" />
  <PackageReference Include="Microsoft.AspNetCore.OpenApi" Version="10.0.10" />
  <PackageReference Include="Microsoft.Extensions.ApiDescription.Server" Version="10.0.10" />
</ItemGroup>
```
Program.cs
```cs
var builder = WebApplication.CreateBuilder(args);
builder.Services.AddApiVersioning().AddOpenApi();
var app = builder.Build();
app.Run();
```
run 
```
dotnet build
```
observe these error
```
Restore complete (0,4s)
  web-net10 net10.0 failed with 31 error(s) (1,0s) → bin\Debug\net10.0\web-net10.dll
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error System.AggregateException: One or more errors occurred. (Could not load file or assembly 'Asp.Versioning.Abstractions, Version=10.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The system cannot find the file specified.)
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error  ---> System.IO.FileNotFoundException: Could not load file or assembly 'Asp.Versioning.Abstractions, Version=10.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The system cannot find the file specified.
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error File name: 'Asp.Versioning.Abstractions, Version=10.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at Asp.Versioning.OpenApi.OpenApiDocumentDescriptionOptions..ctor()
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at Asp.Versioning.OpenApi.OpenApiDocumentDescriptionOptions..ctor()
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at Asp.Versioning.OpenApi.Configuration.VersionedOpenApiOptionsFactory.Create(String name)
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at Asp.Versioning.OpenApi.Configuration.VersionedOpenApiOptionsFactory.CreateAndConfigure(Context newContext)
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at Asp.Versioning.OpenApi.Configuration.ConfigureOpenApiOptions.PostConfigure(String name, OpenApiOptions options)
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at Microsoft.Extensions.Options.OptionsFactory`1.Create(String name)
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at System.Lazy`1.CreateValue()
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at Microsoft.Extensions.Options.OptionsCache`1.GetOrAdd[TArg](String name, Func`3 createOptions, TArg factoryArgument)
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at Microsoft.Extensions.Options.OptionsMonitor`1.Get(String name)
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at Microsoft.AspNetCore.OpenApi.OpenApiDocumentService..ctor(String documentName, IApiDescriptionGroupCollectionProvider apiDescriptionGroupCollectionProvider, IHostEnvironment hostEnvironment, IOptionsMonitor`1 optionsMonitor, IServiceProvider serviceProvider, IServer server)
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at Asp.Versioning.OpenApi.Reflection.Class.OpenApiDocumentService.OpenApiDocumentServiceCtor(String documentName, IApiDescriptionGroupCollectionProvider apiDescriptionGroupCollectionProvider, IHostEnvironment hostEnvironment, IOptionsMonitor`1 optionsMonitor, IServiceProvider serviceProvider, IServer server)
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at Asp.Versioning.OpenApi.Reflection.Class.OpenApiDocumentService.New(IServiceProvider serviceProvider, String documentName)
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at Microsoft.Extensions.DependencyInjection.KeyedServiceContainer.<>c__DisplayClass8_0.<AddService>b__0(IServiceContainer _, Type _)
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at System.ComponentModel.Design.ServiceContainer.GetService(Type serviceType)
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at Microsoft.Extensions.DependencyInjection.KeyedServiceContainer.GetKeyedService(Type serviceType, Object serviceKey)
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at Microsoft.Extensions.DependencyInjection.KeyedServiceContainer.Microsoft.Extensions.DependencyInjection.IKeyedServiceProvider.GetRequiredKeyedService(Type serviceType, Object serviceKey)
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at Microsoft.Extensions.DependencyInjection.ServiceProviderKeyedServiceExtensions.GetRequiredKeyedService[T](IServiceProvider provider, Object serviceKey)
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at Microsoft.Extensions.ApiDescriptions.OpenApiDocumentProvider.GenerateAsync(String documentName, TextWriter writer, OpenApiSpecVersion openApiSpecVersion)
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    --- End of inner exception stack trace ---
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at System.Threading.Tasks.Task.Wait(TimeSpan timeout, CancellationToken cancellationToken)
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at Microsoft.Extensions.ApiDescription.Tool.Commands.GetDocumentCommandWorker.GetDocument(String documentName, String projectName, String outputDirectory, MethodInfo generateMethod, Object service, MethodInfo generateWithVersionMethod, String fileName)
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at Microsoft.Extensions.ApiDescription.Tool.Commands.GetDocumentCommandWorker.GetDocuments(IServiceProvider services)
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error    at Microsoft.Extensions.ApiDescription.Tool.Commands.GetDocumentCommandWorker.Process()
    C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\Microsoft.Extensions.ApiDescription.Server.targets(67,5): error MSB3073: The command "dotnet "C:\Users\redacted\.nuget\packages\microsoft.extensions.apidescription.server\10.0.10\build\../tools/dotnet-getdocument.dll" --assembly "C:\Users\redacted\dev\web-net10\bin\Debug\net10.0\web-net10.dll" --file-list "obj\web-net10.OpenApiFiles.cache" --framework ".NETCoreApp,Version=v10.0" --output "C:\Users\redacted\dev\web-net10\obj" --project "web-net10" --assets-file "C:\Users\redacted\dev\web-net10\obj\project.assets.json" --platform "AnyCPU" " exited with code 11.
```

### Exceptions (if any)

_No response_

### .NET Version

10.0.302

### Anything else?

_No response_

--- a/src/AspNetCore/OData/src/Asp.Versioning.OData.ApiExplorer/Asp.Versioning.OData.ApiExplorer.csproj
+++ b/src/AspNetCore/OData/src/Asp.Versioning.OData.ApiExplorer/Asp.Versioning.OData.ApiExplorer.csproj
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
  <PropertyGroup>
-  <VersionPrefix>10.0.0</VersionPrefix>
+  <VersionPrefix>10.0.1</VersionPrefix>
   <AssemblyVersion>10.0.0.0</AssemblyVersion>
   <TargetFramework>$(DefaultTargetFramework)</TargetFramework>
   <RootNamespace>Asp.Versioning</RootNamespace>

--- a/src/AspNetCore/OData/src/Asp.Versioning.OData/Asp.Versioning.OData.csproj
+++ b/src/AspNetCore/OData/src/Asp.Versioning.OData/Asp.Versioning.OData.csproj
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
  <PropertyGroup>
-  <VersionPrefix>10.0.1</VersionPrefix>
+  <VersionPrefix>10.0.2</VersionPrefix>
   <AssemblyVersion>10.0.0.0</AssemblyVersion>
   <TargetFramework>$(DefaultTargetFramework)</TargetFramework>
   <RootNamespace>Asp.Versioning</RootNamespace>

--- a/src/AspNetCore/WebApi/src/Asp.Versioning.Mvc.ApiExplorer/Asp.Versioning.Mvc.ApiExplorer.csproj
+++ b/src/AspNetCore/WebApi/src/Asp.Versioning.Mvc.ApiExplorer/Asp.Versioning.Mvc.ApiExplorer.csproj
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
  <PropertyGroup>
-  <VersionPrefix>10.0.0</VersionPrefix>
+  <VersionPrefix>10.0.1</VersionPrefix>
   <AssemblyVersion>10.0.0.0</AssemblyVersion>
   <TargetFramework>$(DefaultTargetFramework)</TargetFramework>
   <RootNamespace>Asp.Versioning.ApiExplorer</RootNamespace>

--- a/src/AspNetCore/WebApi/src/Asp.Versioning.Mvc/Asp.Versioning.Mvc.csproj
+++ b/src/AspNetCore/WebApi/src/Asp.Versioning.Mvc/Asp.Versioning.Mvc.csproj
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
  <PropertyGroup>
-  <VersionPrefix>10.0.0</VersionPrefix>
+  <VersionPrefix>10.0.1</VersionPrefix>
   <AssemblyVersion>10.0.0.0</AssemblyVersion>
   <TargetFramework>$(DefaultTargetFramework)</TargetFramework>
   <RootNamespace>Asp.Versioning</RootNamespace>

--- a/src/AspNetCore/WebApi/src/Asp.Versioning.OpenApi/Asp.Versioning.OpenApi.csproj
+++ b/src/AspNetCore/WebApi/src/Asp.Versioning.OpenApi/Asp.Versioning.OpenApi.csproj
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
  <PropertyGroup>
-  <VersionPrefix>10.0.0</VersionPrefix>
+  <VersionPrefix>10.0.1</VersionPrefix>
   <AssemblyVersion>10.0.0.0</AssemblyVersion>
   <TargetFramework>$(DefaultTargetFramework)</TargetFramework>
   <RootNamespace>Asp.Versioning.OpenApi</RootNamespace>

